src/java.base/share/classes/java/lang/foreign/Linker.java

```diff
@@ -409,6 +409,12 @@
  * }
  * }
  *
+ * <p>
+ * It should be noted that values passed as variadic arguments undergo default argument promotion in C. Each value of
+ * type {@code float} is converted to {@code double}, and each integral type smaller than {@code int} is converted to
+ * {@code int}. As such, the native linker will reject attempts to link a function with variadic parameters which have
```

Comment:
I did another pass over this. I went back to the first paragraph and defined the term "variadic argument layout" which I then use to explain which layouts can not be used as variadic argument layouts.

I've also moved the note the be directly after the first paragraph so that it appears closed to where the term is defined.

Hopefully there should be no gaps now.